src/backend/models/flatArray.ts

```diff
@@ -0,0 +1,158 @@
+import Page from './page';
+import PageOrder from './pageOrder';
+import NodeCache from 'node-cache';
+
+// Create cache for flat array
+const cache = new NodeCache({ stdTTL: 120 });
+
+/**
+ * @typedef {object} FlatArrayData
+ * @property {string} id - page id
+ * @property {string} parentId - page parent id
+ * @property {string} rootId - page root id
+ * @property {number} level - page level in sidebar
+ * @property {string} title - page title
+ * @property {string} uri - page uri
+ */
+export interface FlatArrayData {
+  id?: string;
+  parentId?: string;
+  rootId?: string;
+  level?: number;
+  title?: string;
+  uri?: string;
+}
+
+/**
+ * @class FlatArray
+ * @class FlatArray model
+ */
+class FlatArray {
+  /**
+   * Returns pages flat array
+   *
+   * @returns {Promise<Array<FlatArrayData>>}
+   */
+  public static async get(): Promise<Array<FlatArrayData>> {
+    // Get flat array from cache
+    let arr = cache.get('flatArray') as Array<FlatArrayData>;
+
+    // Check is flat array consists in cache
+    if (!arr) {
+      arr = await this.generate();
+    }
+
+    return arr;
+  }
+
+  /**
+   * Generates flat array, saves it to cache, returns it
```

Comment:
describe when we need to call this method. 

Maybe its better to name it `regenerate`?